modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java

```diff
@@ -463,6 +464,14 @@ public ObjectRelationship updateObjectRelationship(
 			indexer.reindex(reverseObjectRelationship);
 		}
 
+		if (StringUtil.equals(
```

Comment:
Hey, @murilostodolni. This if statement will have to check whether the objectFieldId2 actually exists or not before updating "the required", because, if it doesn't exist, NoSuchObjectFieldException will be thrown.

Reply:
Hi @rodrigopaulino, perfect! I agree. I checked in database and in the relationship manyToMany we save the objectFieldId2 with value 0 (zero). So, we need verify if not null and different of the 0 (zero). Are you agree with this?

Reply:
Just check if it's different than zero.